server/content-repo/content-storage.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { storage } = require('../config');
+const Promise = require('bluebird');
+const createStorage = require('../common/storage');
+const values = require('lodash/values');
+
+class ContentStorage {
+  constructor(options = {}) {
+    this.storage = createStorage(options);
+    this.options = options;
+  }
+
+  getRepository(repoId) {
+    const key = `${this._getRepoLocation(repoId)}/index.json`;
+    return this.storage.getItem(key);
+  }
+
+  getCatalog() {
+    const key = `${this.options.publishedContentLocation}/index.json`;
+    return this.storage.getItem(key);
+  }
+
+  getContainer(repoId, id, programId) {
+    const key =
+      `${this._getRepoLocation(repoId, programId)}/${id}.container.json`;
+    return this.storage.getItem(key);
+  }
+
+  getExam(repoId, id, programId) {
+    const key = `${this._getRepoLocation(repoId, programId)}/${id}.exam.json`;
+    return this.storage.getItem(key);
+  }
+
+  getAssessments(repoId, id, programId) {
+    const key =
+      `${this._getRepoLocation(repoId, programId)}/${id}.assessments.json`;
+    return this.storage.getItem(key);
+  }
+
+  resolveContainer(container) {
```

Comment:
What if somebody has array of elements that are not within container (i.e. assessment question). Consider renaming it to resolveElements and process array of elements.